src/renderer/store/lobby.store.ts

```diff
@@ -0,0 +1,288 @@
+    //Recalculate player counts afterward.
+    let maxPlayerCount: number = 0;
+    for (const allyKey in lobbyStore.activeLobby!.allyTeams) {
+        for (const teamKey in lobbyStore.activeLobby!.allyTeams[allyKey]!.teams) {
+            maxPlayerCount += lobbyStore.activeLobby!.allyTeams[allyKey]!.teams[teamKey]!.maxPlayers!;
+        }
+    }
+    lobbyStore.activeLobby!.maxPlayerCount = maxPlayerCount;
+    // TODO: Once spectator type is added, this will not work how we want it to (probably). We will likely have to iterate instead while checking type, queue, etc.
+    const playerCount: number = Object.keys(lobbyStore.activeLobby!.members!).length;
+    lobbyStore.activeLobby!.playerCount = playerCount;
+    // We need to sub to new members, but if members go away we unsub instead. Ugh we need to maintain a list and diff it also lol
+    if (data.members) {
+        const userSubList: UserId[] = [];
+        const userUnsubList: UserId[] = [];
+        for (const memberKey in data.members) {
+            const member = data.members[memberKey];
+            if (member != null) {
+                userSubList.push(member.id);
+            } else {
+                userUnsubList.push(memberKey);
+            }
+        }
+        if (userSubList.length > 0) {
+            window.tachyon.request("user/subscribeUpdates", { userIds: userSubList });
+        }
+        if (userUnsubList.length > 0) {
+            window.tachyon.request("user/unsubscribeUpdates", { userIds: userUnsubList });
+        }
+    }
+}
+
+function onLobbyLeftEvent(data: LobbyLeftEventData) {
+    clearUserSubscriptions();
+    lobbyStore.activeLobby = undefined;
+    console.log("Tachyon event: lobby/left:", data);
+    // If we ever use a specific view for a lobby instead of BattleDrawer, we need to push a route here
+    battleStore.isLobbyOpened = false;
+    notificationsApi.alert({
+        text: i18n.global.t("lobby.multiplayer.custom.removedFromLobby"),
+        severity: "info",
+    });
+}
+
+function clearUserSubscriptions() {
+    const userUnsubList: UserId[] = [];
+    if (lobbyStore.activeLobby?.members) {
+        for (const memberKey in lobbyStore.activeLobby!.members) {
+            if (lobbyStore.activeLobby!.members[memberKey]!.id != me.userId)
+                // Skip unsubbing from ourselves.
+                // TODO: avoid unsubbing from anyone we have subscribe to elsewhere (party, friends?)
+                userUnsubList.push(lobbyStore.activeLobby!.members[memberKey]!.id);
+        }
+        if (userUnsubList.length > 0) {
+            window.tachyon.request("user/unsubscribeUpdates", { userIds: userUnsubList });
+        }
+    }
+}
+
+function clearLobbyAndListInfo() {
+    lobbyStore.lobbyList = {};
+    lobbyStore.selectedLobby = null;
+    lobbyStore.activeLobby = undefined;
+}
+
+export async function initLobbyStore() {
+    window.tachyon.onEvent("lobby/left", (data) => {
```

Comment:
nit pick: it could be just `window.tachyon.onEvent("lobby/left", onLobbyLeftEvent)`